middleware/statsd/statsd_middleware.py

```diff
@@ -0,0 +1,50 @@
+import os
+import time
+
+from datadog import DogStatsd
+from django.utils.deprecation import MiddlewareMixin
+
+
+statsd_host = os.environ.get("STATSD_ENDPOINT")
+statsd_port = int(os.environ.get("STATSD_PORT"))
+statsd = DogStatsd(host=statsd_host, port=statsd_port)
+
+REQUEST_LATENCY_METRIC_NAME = "django_request_latency_seconds"
+REQUEST_COUNT_METRIC_NAME = "django_request_count"
+
+
+class StatsdMetricsMiddleware(MiddlewareMixin):
+    def process_request(self, request):
+        request.start_time = time.time()
+
+    def get_view_name(self, request):
+        view_name = "<unnamed view>"
+        if hasattr(request, "resolver_match"):
+            if request.resolver_match is not None:
+                if request.resolver_match.view_name is not None:
+                    view_name = request.resolver_match.view_name
+        return view_name
+
+    def process_response(self, request, response):
+        statsd.increment(
+            REQUEST_COUNT_METRIC_NAME,
+            tags=[
+                "service:django_worker",
+                "method:%s" % request.method,
```

Comment:
@burnerlee why do we have to specify `method` text in the tag? Wouldn't this change the way queries will be written?